### PR TITLE
AK/#2598 freespeed graph

### LINKED ISSUE #2598: add % of free speed graph in root folder

### PR DESCRIPTION
**Pull request by YOxan**
Closes #2598 

This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/2622)

### CODE REVIEW

**Comment by YOxan** [comment]
test!

File: src/main/scala/beam/analysis/cartraveltime/CarTripStatsFromPathTraversalEventHandler.scala [hunk 1]
```diff
@@ -144,6 +148,7 @@ class CarTripStatsFromPathTraversalEventHandler(
     createIterationGraphForAverageSpeed(dataSet, iterationNumber, mode)
   }
 
+  //TODO: maybe collect all Iteration end events and then at the end compute an info for all iterations
```
Comment (REASY) [inline src/main/scala/beam/analysis/cartraveltime/CarTripStatsFromPathTraversalEventHandler.scala:151]: If you would do it only once in the end, then you would be able to review the graph and CSV files only in the very end. For production runs one iteration takes around 1 hour and usually we run 15-20 iterations, so all this time you won't have any idea what happens with simulation.

File: src/main/scala/beam/analysis/cartraveltime/CarTripStatsFromPathTraversalEventHandler.scala [hunk 2]
```diff
@@ -159,10 +164,19 @@ class CarTripStatsFromPathTraversalEventHandler(
       getIterationCarRideStats(event.getIteration, singleRideStats)
     }
 
+    iterationsCarTripInfo ++= type2Statistics.map {
+      case (carType, iterationCarTripStats) => (event.getIteration, carType) -> iterationCarTripStats
+    }
+
     averageCarSpeedPerIterationByType += type2Statistics.mapValues(_.speed.stats.avg)
 
+    //FIXME: create chart and store the image every time iteration ends... Why so?
```
Comment (REASY) [inline src/main/scala/beam/analysis/cartraveltime/CarTripStatsFromPathTraversalEventHandler.scala:173]: This is expected behavior, we re-created the graph in the end of every iteration. Over the iterations the graph contains more data point

File: src/main/scala/beam/analysis/cartraveltime/CarTripStatsFromPathTraversalEventHandler.scala [hunk 3]
```diff
@@ -159,10 +164,19 @@ class CarTripStatsFromPathTraversalEventHandler(
...
     createRootGraphForAverageCarSpeedByType(event)
 
+    if (event.getIteration == lastIterationNumber) {
```
Comment (REASY) [inline src/main/scala/beam/analysis/cartraveltime/CarTripStatsFromPathTraversalEventHandler.scala:176]: I'd suggest to follow the same patter as for other graphs and re-create the graph on every iteration.

Comment (YOxan) [reply src/main/scala/beam/analysis/cartraveltime/CarTripStatsFromPathTraversalEventHandler.scala]: Done

**Comment by YOxan** [comment]
test!

**Pull request merged**